### PR TITLE
Prime

### PR DESCRIPTION
**Pull request by jordanmckinney**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,6 +7,9 @@
     "react-dom": "^15.5.4"
   },
   "devDependencies": {
+    "babel-eslint": "^7.2.3",
```
Comment (nvandoorn) [inline package.json]: Did you reject the react scripts or something? I believe these dependencies are usually covered by the react scripts

File: src/App.js [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
 
 
 const App = () => {
   return (
     <div className="App">
-      <Counter base="2"/>
-      <Counter base="3"/>
-      <Counter base="6"/>
-      <Counter base="8"/>
+      <Header/>
```
Comment (nvandoorn) [inline src/App.js]: you can omit the `{}` and `return` (`const myFunc = () => 'return value'`)

File: src/component_style/Footer.css [hunk 3]
```diff
@@ -0,0 +1,8 @@
+.Footer {
```
Comment (nvandoorn) [inline src/component_style/Footer.css]: Use kebab-case for HTML/CSS classes and IDs as these are not case sensitive.

File: src/components/Calc.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React, { Component } from 'react';
+import '../component_style/Calc.css';
+import Powers from './Powers';
+import Primes from './Primes';
+
+class Body extends Component {
```
Comment (nvandoorn) [inline src/components/Calc.js]: Any reason this is named `Body` instead of `Calc`?

File: src/components/Footer.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import React, { Component } from 'react';
+import '../component_style/Footer.css';
+
+class Footer extends Component {
```
Comment (nvandoorn) [inline src/components/Footer.js]: This component doesn't have any state so it can probably be a function

File: src/components/Powers.js [hunk 6]
```diff
@@ -1,29 +1,29 @@
...
     super(props);
-    this.state = { result: 1, power: 0 };
+    this.state = { result: 1, exponent: 0 };
     this.increment = this.increment.bind(this);
     this.reset = this.reset.bind(this);
   }
 
   increment() {
+    var current_exponent = this.state.exponent;
+    this.setState({ exponent: current_exponent += 1 });
```
Comment (nvandoorn) [inline src/components/Powers.js]: You can use `setState` with a callback which may clean this up a bit. `this.setState(prevState => { prevState.exponent++; return prevState })` could work

File: src/helpers/primeSieve.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+const primeSieve = (limit) => {
+  var sieve = [], i, j, primesList = [];
```
Comment (nvandoorn) [inline src/helpers/primeSieve.js]: Any reason var was selected over `let`? Not really a problem but `let` scoping is tighter.

File: src/helpers/primeSieve.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+const primeSieve = (limit) => {
```
Comment (nvandoorn) [inline src/helpers/primeSieve.js]: Probably a little big to be an arrow function. Consider use the `function` keyword.

File: src/components/Calc.js [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+  render() {
+    return (
+      <div className="Body">
+        <Powers base="2"/>
+        <Powers base="3"/>
+        <Primes initial_limit="10"/>
```
Comment (nvandoorn) [inline src/components/Calc.js]: I believe convention is to camelCase props but not a huge deal

File: package.json (hunk 1 above)
Comment (jordanmckinney) [inline package.json]: So I shouldn't have to add these right? These should be included already?

File: src/components/Calc.js (hunk 4 above)
Comment (jordanmckinney) [inline src/components/Calc.js]: changed it back!

File: src/components/Calc.js (hunk 9 above)
Comment (jordanmckinney) [inline src/components/Calc.js]: hmm yea I mostly copied the styling that create-react-app gave me, but I'm not so sure about some of it

File: package.json (hunk 1 above)
Comment (nvandoorn) [inline package.json]: I believe the `react-scripts` dependency has includes these as dependencies

File: src/components/Calc.js (hunk 9 above)
Comment (nvandoorn) [inline src/components/Calc.js]: Where did you see this in create-react-app?

File: src/helpers/primeSieve.js (hunk 7 above)
Comment (jordanmckinney) [inline src/helpers/primeSieve.js]: done (I copied it from SO)

File: src/calc/calc.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
+import React, { Component } from 'react';
+import './calc.css';
+import Powers from '../powers/powers';
+import Primes from '../primes/primes';
+
+class Calc extends Component {
```
Comment (nvandoorn) [inline src/calc/calc.js]: Do you plan on adding state to these components? If not, probably worth converting to a function.

**Pull request merged**